sertop/sertop_sexp.ml

```diff
@@ -160,10 +160,14 @@ let ser_loop ser_opts =
 
   (* Main loop *)
   let rec loop cmd_id =
-    try
-      let cmd_tag, cmd = read_cmd cmd_id ser_opts.in_chan pp_err          in
-      pp_ack cmd_tag;
-      iter pp_answer @@ map (fun a -> SP.Answer (cmd_tag, a)) (SP.exec_cmd cmd);
-      if not (is_cmd_quit cmd) then loop (1+cmd_id)
-    with Sys.Break -> loop (1+cmd_id)
+    let quit =
+      try
+        let cmd_tag, cmd = read_cmd cmd_id ser_opts.in_chan pp_err          in
+        pp_ack cmd_tag;
+        iter pp_answer @@ map (fun a -> SP.Answer (cmd_tag, a)) (SP.exec_cmd cmd);
+        is_cmd_quit cmd
+      with
+        Sys.Break -> false
+    in if not quit then loop (1+cmd_id) else ()
```

Comment:
There is still a very small chance that you can make SerAPI exit here due to sending `Break` too fast.

Reply:
perhaps we can accept this tradeoff against blowing up stack if the main loop is not tail recursive ?

Reply:
You could wrap the body of the `try` in a second loop, and then you'd only blow pay for one stack entry per Sys.Break instead of one per command maybe?

Reply:
@cpitclaudel I tried a couple of modifications but it seems that the a short instruction sequence like (cmp, je, inc, call) that is executed when Sys.Break signal is caught anyways would not be protected from another Sys.Break signal. In the original code the instruction sequence (inc, call) executed in the line 168 underlying "loop (1 + cmd_id)" also isn't  protected from another Sys.Break signal, is it?  If you know how to improve, may be drop a line pls, or otherwise shall we merge this patch as is?

Reply:
Indeed I think it is OK; to be picky we could call the loop from inside the exception handler.